safehttp/plugins/fetch_metadata/fetch_metadata_test.go

```diff
@@ -0,0 +1,407 @@
+			method: safehttp.MethodPost,
+			site:   "cross-site",
+			mode:   "cors",
+			dest:   "document",
+		},
+		{
+			name:   "cross origin no cors",
+			method: safehttp.MethodPost,
+			site:   "cross-site",
+			mode:   "no-cors",
+			dest:   "nested-document",
+		},
+	}
+)
+
+func TestAllowedResourceIsolationEnforceMode(t *testing.T) {
+	tests := allowedRIPHeaders
+	for _, test := range allowedRIPNavHeaders {
+		tests = append(tests, testHeaders{
+			name:   test.name,
+			method: test.method,
+			site:   test.site,
+			mode:   test.mode,
+			dest:   test.dest,
+		})
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			req := safehttptest.NewRequest(test.method, "https://spaghetti.com/carbonara", nil)
+			req.Header.Add("Sec-Fetch-Site", test.site)
+			req.Header.Add("Sec-Fetch-Mode", test.mode)
+			req.Header.Add("Sec-Fetch-Dest", test.dest)
+			rec := safehttptest.NewResponseRecorder()
+
+			p := fetchmetadata.NewPlugin()
+			p.Before(rec.ResponseWriter, req)
+
+			if want, got := safehttp.StatusOK, safehttp.StatusCode(rec.Status()); got != want {
+				t.Errorf("status code got: %v want: %v", got, want)
+			}
+			if diff := cmp.Diff(map[string][]string{}, map[string][]string(rec.Header())); diff != "" {
+				t.Errorf("rec.Header() mismatch (-want +got):\n%s", diff)
+			}
+			if want, got := "", rec.Body(); got != want {
+				t.Errorf("response body got: %q want: %q", got, want)
+			}
+		})
+	}
+}
+
+func TestRejectedResourceIsolationEnforceMode(t *testing.T) {
+	tests := disallowedRIPHeaders
+	for _, test := range disallowedRIPNavHeaders {
+		tests = append(tests, testHeaders{
+			name:   test.name,
+			method: test.method,
+			site:   test.site,
+			mode:   test.mode,
+			dest:   test.dest,
+		})
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			req := safehttptest.NewRequest(test.method, "https://spaghetti.com/carbonara", nil)
+			req.Header.Add("Sec-Fetch-Site", test.site)
+			req.Header.Add("Sec-Fetch-Mode", test.mode)
+			req.Header.Add("Sec-Fetch-Dest", test.dest)
+			rec := safehttptest.NewResponseRecorder()
+
+			p := fetchmetadata.NewPlugin()
+			p.Before(rec.ResponseWriter, req)
+
+			if want, got := safehttp.StatusForbidden, safehttp.StatusCode(rec.Status()); want != got {
+				t.Errorf("status code got: %v want: %v", got, want)
+			}
+			wantHeaders := map[string][]string{
+				"Content-Type":           {"text/plain; charset=utf-8"},
+				"X-Content-Type-Options": {"nosniff"},
+			}
+			if diff := cmp.Diff(wantHeaders, map[string][]string(rec.Header())); diff != "" {
+				t.Errorf("rec.Header() mismatch (-want +got):\n%s", diff)
+			}
+			if want, got := "Forbidden\n", rec.Body(); got != want {
+				t.Errorf("response body got: %q want: %q", got, want)
+			}
+		})
+	}
+}
+
+type fooLog struct {
+	report string
+}
+
+func (l *fooLog) Log(r *safehttp.IncomingRequest) {
+	l.report = r.Method()
+}
+func TestAllowedResourceIsolationReportMode(t *testing.T) {
```

Comment:
You have `TestAllowedResourceIsolationReportMode` and `TestAllowedResourceIsolationEnforceMode`. But having `ReportOnly` turned on doesn't make a difference for requests that are allowed to pass. It only changes the behavior for requests that are disallowed. Therefore I would say you could remove one of these table tests without losing any test coverage.

Reply:
I still want to make sure that nothing was written to `logger.report`